src/theory/quantifiers/sygus_grammar_norm.cpp

```diff
@@ -345,7 +339,7 @@ SygusGrammarNorm::Transf* SygusGrammarNorm::inferTransf(
     Kind actual_sop_k = nm->operatorToKind(sop);
     if (sop.getKind() == BOUND_VARIABLE
         || ((sop.getKind() != BUILTIN || actual_sop_k == ITE) && sop.isConst()
-            && !TransfChain::isId(sygus_tn, PLUS, Node::fromExpr(sop))))
+            && !TransfChain::isId(sygus_tn, nm->operatorOf(PLUS), Node::fromExpr(sop))))
```

Comment:
This condition should not depend on anything other than isId.

Reply:
By making the changes we discussed (i.e. ``MINUS`` gets its own monomial) one of the regressions does not goes through anymore. It is "process-10-vars-2fun.sy", whose solution needs a sum of three occurrences of a variable. Looking at the trace the enumeration is getting stuck in a series of applications of ``MINUS``, which is getting the first monomial in the chain.